File: app/src/main/java/au/id/blackwell/kurt/lantv/resolver/MediaResolverFactory.java
package au.id.blackwell.kurt.lantv.resolver;

import android.net.Uri;
import android.webkit.WebView;

import au.id.blackwell.kurt.lantv.MediaDetails;
import au.id.blackwell.kurt.lantv.utility.Pool;

public final class MediaResolverFactory {
    Pool<WebView> mWebViewPool;

    public MediaResolverFactory(Pool<WebView> webViewPool) {
        mWebViewPool = webViewPool;
    }

    public MediaResolver create(Uri mediaResolveUri) {
        switch (mediaResolveUri.getScheme()) {
            case "cctv":
                return new CctvMediaResolver(mWebViewPool, mediaResolveUri.toString().replaceFirst("^cctv://", "http://"));
            default:
                return new StaticMediaResolver(new MediaDetails(mediaResolveUri));
        }
    }
}

File: app/src/main/java/au/id/blackwell/kurt/lantv/resolver/MediaResolverFactory.kt
package au.id.blackwell.kurt.lantv.resolver

import android.net.Uri
import android.webkit.WebView

import au.id.blackwell.kurt.lantv.MediaDetails
import au.id.blackwell.kurt.lantv.utility.Pool

class MediaResolverFactory(internal var mWebViewPool: Pool<WebView>) {

    fun create(mediaResolveUri: Uri): MediaResolver {
        when (mediaResolveUri.scheme) {
            "cctv" -> return CctvMediaResolver(mWebViewPool, mediaResolveUri.toString().replaceFirst("^cctv://".toRegex(), "http://"))
            else -> return StaticMediaResolver(MediaDetails(mediaResolveUri))
        }
    }
}
